otter/controller.py

```diff
@@ -303,6 +302,31 @@ def modify_state(group_info):
     return d
 
 
+@defer.inlineCallbacks
+def modify_and_trigger(dispatcher, group, logargs, modifier, *args, **kwargs):
+    """
+    Modify group state and trigger convergence after that
+
+    :param IScalingGroup group: Scaling group whose state is getting modified
+    :param log: Bound logger
+    :param modifier: Callable as described in IScalingGroup.modify_state
+
+    :return: Deferred with None
+    """
+    cannot_exec_pol_err = None
+    try:
+        yield group.modify_state(modifier, *args, **kwargs)
+    except CannotExecutePolicyError as ce:
+        cannot_exec_pol_err = ce
+    if tenant_is_enabled(group.tenant_id, config_value):
+        eff = Effect(
+            BoundFields(
+                trigger_convergence(group.tenant_id, group.uuid), logargs))
+        yield perform(dispatcher, eff)
+    if cannot_exec_pol_err is not None:
```

Comment:
Slightly more context for this currently expected behavior:  We don't distinguish between the two following cases of `CannotExecutePolicyError` on `delta=0`: 

1.  The user is already at their max, and hence can't scale up.  (similarly with min)
2.  The user tries to execute a policy which sets the desired capacity to the current desired capacity.

I think it makes sense in the first case to not converge.  I can see wanting to converge in the second case.

(I'm not arguing one way or another - @manishtomar and I discussed just removing that test expectation instead, but I think we decided it'd be easier to keep current functionality for now, and if we decided to change the behavior, do it in a different PR).